### PR TITLE
[ISSUE #4263] InterruptedExceptions should never be ignored[ThreadUtils]

### LINKED ISSUE #4263: [Enhancement] InterruptedExceptions should never be ignored[ThreadUtils].
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
located at:
eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/ThreadUtils.java line 36,53
analysis and explanation:
InterruptedExceptions should never be ignored in the code, and simply logging the exception counts in this case as "ignoring". The throwing of the InterruptedException clears the interrupted state of the Thread, so if the exception is not handled properly the information that the thread was interrupted will be lost. Instead, InterruptedExceptions should either be rethrown - immediately or after cleaning up the method’s state - or the thread should be re-interrupted by calling Thread.interrupt() even if this is supposed to be a single-threaded application. Any other course of action risks delaying thread shutdown and loses the information that the thread was interrupted - probably without finishing its task.

### Describe the solution you'd like

Restore interrupted state
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by harshithasudhakar**
Fixes #4263 

### Modifications

Restored interrupted states in lines 37 & 54

### Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/ThreadUtils.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public static void randomPause(long min, long max, TimeUnit timeUnit) {
             long timeout = ThreadLocalRandom.current().nextLong(min, max + 1);
             timeUnit.sleep(timeout);
         } catch (InterruptedException ignore) {
-            //ignore
+            Thread.currentThread().interrupt;
```
Comment (Alonexc) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/ThreadUtils.java:37]:
```
Thread.currentThread().interrupt();
```

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/ThreadUtils.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public static void sleep(long timeout, TimeUnit timeUnit) {
         try {
             sleepWithThrowException(timeout, timeUnit);
         } catch (InterruptedException ignore) {
-            //ignore
+            Thread.currentThread().interrupt;
```
Comment (Alonexc) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/ThreadUtils.java:54]: same as above.

**Review by Pil0tXia — APPROVED** [review]
LGTM

**Review by mxsm — APPROVED** [review]
LGTM

**Review by Alonexc — APPROVED** [review]

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**